ice/api/utils.py

```diff
@@ -1,4 +1,17 @@
+import inspect
+import os
+import sys
+
 from ice.llutil.launcher.launcher import _parse_devices_and_backend
 
+
 def parse_devices(devices:str):
-    return _parse_devices_and_backend(devices)[0]
\ No newline at end of file
+    return _parse_devices_and_backend(devices)[0]
+
+# def caller_file():
+#     return os.path.abspath(inspect.stack()[1][1])
+
+def extend_sys_path(relpath):
```

Comment:
This function is a general python util function. You can use it to insert new path into `sys.path` by passing a parameter relative to the caller's file path.